src/check_result.jl

```diff
@@ -0,0 +1,51 @@
+# For once you have the sensitivity by two methods (e.g  both finite-differencing and  AD)
+# the code here checks it is correct.
+# Goal is to only call `@isapprox` on things that render well
+
+"""
+    check_equal(actual, expected; kwargs...)
+
+`@test`'s  that `actual ≈ expected`, but breaks up data such that human readable results
+are shown on failures.
+All keyword arguments are passed to `isapprox`.
+"""
+function check_equal(
+    actual::Union{AbstractArray{<:Number}, Number},
+    expected::Union{AbstractArray{<:Number}, Number};
+    kwargs...
+)
+    @test isapprox(actual, expected; kwargs...)
+end
+
+function check_equal(actual::AbstractThunk, expected; kwargs...)
+    check_equal(unthunk(actual), expected; kwargs...)
+end
+
+
+function check_equal(
+    actual::Union{Composite, AbstractArray},
```

Comment:
The `AbstractArray` here is to catch weird things like 
`Vector{Matrix{Float64}}` which we do actually have.

Reply:
done